core/trino-main/src/main/java/io/trino/security/AccessControlManager.java

```diff
@@ -1150,6 +1156,41 @@ public void checkCanExecuteFunction(SecurityContext securityContext, QualifiedOb
                 (control, context) -> control.checkCanExecuteFunction(context, functionName.asSchemaRoutineName()));
     }
 
+    @Override
+    public void checkCanExecuteFunction(SecurityContext securityContext, QualifiedObjectName functionName, FunctionKind functionKind)
+    {
+        requireNonNull(securityContext, "securityContext is null");
+        requireNonNull(functionName, "functionName is null");
+
+        checkCanAccessCatalog(securityContext, functionName.getCatalogName());
+
+        systemAuthorizationCheck(control -> control.checkCanExecuteFunction(
+                securityContext.toSystemSecurityContext(),
+                functionName.asCatalogSchemaRoutineName(),
+                fromFunctionKind(functionKind)));
+
+        catalogAuthorizationCheck(
+                functionName.getCatalogName(),
+                securityContext,
+                (control, context) -> control.checkCanExecuteFunction(context, functionName.asSchemaRoutineName(), fromFunctionKind(functionKind)));
+    }
+
+    private static FunctionType fromFunctionKind(FunctionKind functionKind)
+    {
+        switch (functionKind) {
+            case SCALAR:
+                return SCALAR;
+            case AGGREGATE:
+                return AGGREGATE;
+            case WINDOW:
+                return WINDOW;
+            case TABLE:
+                return TABLE;
+            default:
```

Comment:
remove `default:` and move `throw` out of the `switch`. The error prone should fail the build in such case when enum is not enumerated fully with such code. Please apply it to other places.

Reply:
Thanks for the hint! The conversions are no more needed, since `FunctionKind` is moved to SPI.